Core/automation/lib/python/core/date.py

```diff
@@ -118,11 +138,22 @@ def to_java_zoneddatetime(value):
     if isinstance(value, (LegacyDateTimeType, DateTimeType)):
         return to_java_zoneddatetime(value.calendar)
 
-    raise Exception("Invalid conversion: " + str(type(value)))
+    raise TypeError("Unknown type: " + str(type(value)))
```

Comment:
This is the only module change that I found. It would be better in a separate PR, but I'm OK letting this one slip through.

Reply:
Where is this coming from? I think that type error was in my rewrite of it which should have been before I made this branch...

Reply:
Current code is Exception, so if you changed it before, it never made it in... https://github.com/openhab-scripters/openhab-helper-libraries/blob/master/Core/automation/lib/python/core/date.py#L121

Reply:
Hmm I wonder if I missed one when I updated it. The generic except was too vague so I changed it to type error. We could make another commit in here to fix it silently. This whole PR is going to get squashed anyway

Reply:
Oh... now I see. I was just looking at this quickly on my phone earlier. I must have come across this while writing the doc for that function and changed it to be a more specific error. Should I take it out?

Reply:
That would be the 'right' thing to do, but I'm OK letting this go through. I've been working on documentation all day and hope to get a PR in to your repo tonight.

Reply:
I agree about the semantics, but I think it would be more trouble to take out than it's worth.